pallets/sidechain/src/lib.rs

```diff
@@ -40,6 +56,11 @@ pub mod pallet {
 	pub(super) type SlotsPerEpoch<T: Config> =
 		StorageValue<_, sidechain_slots::SlotsPerEpoch, ValueQuery>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn last_mc_hash)]
+	pub(super) type LastMcHash<T: Config> =
+		StorageValue<_, sidechain_domain::McBlockHash, OptionQuery>;
```

Comment:
Would changing it to value query and adding a requirement to set it in genesis simplify things?